### ISSUE TITLE
[clang] False positive -Wmissing-variable-declarations with constexpr variable template instantiation

### ISSUE THREAD

**Issue by carlosgalvezp** [llvm/llvm-project#193937]
Example:

```cpp
template <typename T>
struct Foo{};

template <typename T>
constexpr auto x = Foo<T>{};

constexpr auto y = x<int>; // NOK

constexpr int z = 123;  // OK
```

Gives:

```
<source>:5:16: warning: no previous extern declaration for non-static variable 'x<int>' [-Wmissing-variable-declarations]
    5 | constexpr auto x = Foo<T>{};
      |                ^
<source>:7:20: note: in instantiation of variable template specialization 'x' requested here
    7 | constexpr auto y = x<int>;
      |                    ^
<source>:5:11: note: declare 'static' if the variable is not intended to be used outside of this translation unit
    5 | constexpr auto x = Foo<T>{};
      |           ^
1 warning generated.
```
[Godbolt](https://godbolt.org/z/4oehEsv5h).

`constexpr` variables are implicitly `static`, so the check should not warn here. 

PS: it's also strange that it warns on line 5 instead of line 7, since it complains about `x<int>`.

**Comment by carlosgalvezp** [comment]
The issue goes away when not using `auto`, so I think it's a duplicate of https://github.com/llvm/llvm-project/issues/41387
